lib/experimental/kses.php

```diff
@@ -65,3 +65,22 @@ function gutenberg_override_core_kses_init_filters() {
 // The 'kses_init_filters' is usually initialized with default priority. Use higher priority to override.
 add_action( 'init', 'gutenberg_override_core_kses_init_filters', 20 );
 add_action( 'set_current_user', 'gutenberg_override_core_kses_init_filters' );
+
+/**
+ * See https://github.com/WordPress/wordpress-develop/pull/4108
+ *
+ * Mark CSS safe if it contains a "filter: url('#wp-duotone-...')" rule.
+ *
+ * This function should not be backported to core.
+ *
+ * @param bool   $allow_css Whether the CSS is allowed.
+ * @param string $css_test_string The CSS to test.
+ */
+function allow_filter_in_styles( $allow_css, $css_test_string ) {
+	if ( strpos( $css_test_string, "filter: url('#wp-duotone" ) !== false ) {
```

Comment:
Question: does this mean I could create a CSS string that included `"filter: url('#wp-duotone"` in order to allow _any_ CSS rule to bypass the filtering rules.

Assume we'll want to ditch this asap in order to get the Core patch landed?

Reply:
Yeah we should

Reply:
yes, but it's not in order to get the core patch landed. the core patch should land for the reasons described by @scruffian in the ticket. This is here so we can work on this without core `trunk`

Reply:
In core, when a url attribute is used, the value is parsed with some more complex regex (here: https://github.com/WordPress/wordpress-develop/blob/365efb13c2f2f9de572e53a4c115a0d59232e54f/src/wp-includes/kses.php#L2525). Would it be safer to copy + paste the regexes from there so that we can ensure the same level of parsing occurs for the the `filter:` rule?

Reply:
```suggestion
	if ( preg_match( "^filter: url\('#wp-[-a-zA-Z0-9]+'\);$", $css_test_string ) ) {
```

Might be a bit safer to use a a regex and only allow 100% correct urls rather than just a strpos check

Reply:
That sounds like an elegant way to do it 👍

Reply:
I didn't test that regex, so it might need some tweaks if it isn't processing line by line or newlines have already been stripped.

Reply:
Ah, good point. I think the string in `preg_match` will need a`/` delimiter at the beginning and end, too, to flag that it's a pattern, but overall I think that's a better and simpler way to do it than copying the more complex regex from core.

Reply:
Hm with a  good preg match we could in theory even live with this in the plugin for a while.

Reply:
Added in [e1ab155](https://github.com/WordPress/gutenberg/pull/48281/commits/e1ab155bab043303f23954a6d4d6d6d34bb76b9c)